UnityProject/Assets/Scripts/Learning/ProtipManager.cs

```diff
@@ -70,26 +62,20 @@ private void CheckQueue()
 		{
 			if(IsShowingTip || queuedTips.Count == 0) return;
 			var tip = queuedTips.Dequeue();
-			ShowTip(tip.Text, tip.Duration, tip.Sprite, tip.Animation);
+			ShowTip(tip);
 		}
 
-		public void QueueTip(string TipText, float duration = 25f, Sprite img = null, ProtipUI.SpriteAnimation showAnim = ProtipUI.SpriteAnimation.ROCKING)
+		public void QueueTip(ProtipSO tip)
 		{
-			ProTipUIData newEntry = new ProTipUIData
-			{
-				Text = TipText,
-				Duration = duration,
-				Sprite = img,
-				Animation = showAnim
-			};
-			queuedTips.Enqueue(newEntry);
+			queuedTips.Enqueue(tip);
 		}
 
-		private void ShowTip(string TipText, float duration = 25f, Sprite img = null, ProtipUI.SpriteAnimation showAnim = ProtipUI.SpriteAnimation.ROCKING)
+		private void ShowTip(ProtipSO tip)
 		{
-			if(duration <= 0) duration = 25f; //Incase whoever was setting the SO data forgot to set the duration.
+			var duration = tip.TipData.ShowDuration;
+			if(tip.TipData.ShowDuration <= 0) duration = 25f; //Incase whoever was setting the SO data forgot to set the duration.
```

Comment:
```
var duration = tip.TipData.ShdowDuration > 0 ? tip.TipData.ShowDuration : 25f
```
except that ``25f`` should be a constant in the script.

Reply:
done